providers/hns/hns_roce_u_hw_v1.c

```diff
@@ -471,19 +470,20 @@ static int hns_roce_u_v1_post_send(struct ibv_qp *ibvqp, struct ibv_send_wr *wr,
 	struct hns_roce_wqe_data_seg *dseg = NULL;
 	struct hns_roce_qp *qp = to_hr_qp(ibvqp);
 	struct hns_roce_context *ctx = to_hr_ctx(ibvqp->context);
+	unsigned int wqe_idx;
 
 	pthread_spin_lock(&qp->sq.lock);
 
-	/* check that state is OK to post send */
-	ind = qp->sq.head;
-
 	for (nreq = 0; wr; ++nreq, wr = wr->next) {
 		if (hns_roce_wq_overflow(&qp->sq, nreq,
 					 to_hr_cq(qp->ibv_qp.send_cq))) {
 			ret = -1;
 			*bad_wr = wr;
 			goto out;
 		}
+
+		wqe_idx = (qp->sq.head + nreq) & (qp->sq.wqe_cnt - 1);
+
```

Comment:
I didn't understand your problem and solution, but a couple of lines above you have hns_roce_wq_overflow() function which supposes to protect from overflow.

Reply:
plus the "a %= b" construction is the better way to write it.

Reply:
> I didn't understand your problem and solution, but a couple of lines above you have hns_roce_wq_overflow() function which supposes to protect from overflow.
Hi, Leon, thanks. You are right, the commit message is uncorrect, I will rewrite it.

Reply:
> I didn't understand your problem and solution, but a couple of lines above you have hns_roce_wq_overflow() function which supposes to protect from overflow.

Hi, Leon, thanks. You are right, the commit message is uncorrect, I will rewrite it.

Reply:
> plus the "a %= b" construction is the better way to write it.

Hi Leon, do you mean: 
wqe_idx = (qp->sq.head + nreq)  %  qp->sq.wqe_cnt;